api/layer/container.go

```diff
@@ -207,6 +207,43 @@ func (n *layer) waitEACLPresence(ctx context.Context, cid *cid.ID, table *eacl.T
 		return fmt.Errorf("couldn't marshal eacl: %w", err)
 	}
 
+	return waitFor(ctx, params, func(ctx context.Context) bool {
+		eaclTable, err := n.pool.GetEACL(ctx, cid)
+		if err == nil {
+			got, err := eaclTable.Marshal()
+			if err == nil && bytes.Equal(exp, got) {
+				return true
+			}
+		}
+		return false
+	})
+}
+
+func (n *layer) deleteContainer(ctx context.Context, cid *cid.ID) error {
+	var sessionToken *session.Token
+	boxData, err := GetBoxData(ctx)
+	if err == nil {
+		sessionToken = boxData.Gate.SessionTokenForDelete()
+	}
+
+	if err = n.pool.DeleteContainer(ctx, cid, pool.WithSession(sessionToken)); err != nil {
+		return err
+	}
+
+	return n.waitForContainerRemoved(ctx, cid, defaultWaitParams())
+}
+
+func (n *layer) waitForContainerRemoved(ctx context.Context, cid *cid.ID, params *waitParams) error {
```

Comment:
Minor suggestion: rename `cid` parameter in order to no collide with package name (in `deleteContainer` too).